modules/control/controller/mpc_controller.cc

```diff
@@ -625,16 +666,39 @@ void MPCController::ComputeLongitudinalErrors(
 
   ADEBUG << "matched point:" << matched_point.DebugString();
   ADEBUG << "reference point:" << reference_point.DebugString();
-  debug->set_station_error(reference_point.path_point().s() - s_matched);
-  debug->set_speed_error(reference_point.v() - s_dot_matched);
+
+  double heading_error = common::math::NormalizeAngle(
+      VehicleStateProvider::Instance()->heading() - matched_point.theta());
+  double lon_speed = VehicleStateProvider::Instance()->linear_velocity() *
```

Comment:
Make pointer or reference for long names, especially if they are used more than once.
`auto* linear_velocity = VehicleStateProvider::Instance()->linear_velocity();`
Shorter lines are more readable.